web/js/map-OpenLayers.js

```diff
@@ -1093,17 +1093,36 @@ OpenLayers.Control.PermalinkFMS = OpenLayers.Class(OpenLayers.Control.Permalink,
 
         if ('replaceState' in history) {
             if (fixmystreet.page.match(/around|reports/)) {
-                history.replaceState(
-                    history.state,
-                    null,
-                    href
-                );
+                try {
+                    history.replaceState(
+                        history.state,
+                        null,
+                        href
+                    );
+                } catch (e) {
+                    // This should rarely happen, thanks to the debounced call
+                    // to _updateLink, but belt and braces.
+                }
             }
         }
     },
     updateLink: function() {
         this._updateLink(0);
     },
+    initialize: function() {
+        OpenLayers.Control.Permalink.prototype.initialize.apply(this, arguments);
+
+        // Zoom events on iOS/Android generate a huge number of calls to
+        // updateLink each of which calls replaceState. On iOS calling
+        // replaceState too often soon results in subsequent calls
+        // raising a "SecurityError: Attempt to use history.replaceState()
+        // more than 100 times per 30 seconds" exception, which results
+        // in the map breaking.
+        // Debouncing the call to _updateLink works around this problem.
+        // A better solution would be to stop so many updateLink calls
```

Comment:
Turns out this is because zooming in is causing a lot more layers to be in range so they each fire a changelayer event which the permalink is listening for. It doesn't need to listen for those events (or the changebaselayer either), which I think should fix this.